apps/media-viewer/src/mixins/loader.js

```diff
@@ -0,0 +1,138 @@
+import { mapActions, mapGetters } from 'vuex'
+import queryString from 'query-string'
+import { basename, dirname } from 'path'
+
+// TODO: this file is a first attempt to separate file/folder loading logic out of the mediaviewer
+// Discussion how to progress from here can be found in this issue:
+// https://github.com/owncloud/phoenix/issues/3301
+
+export default {
+  computed: {
+    ...mapGetters('Files', ['publicLinkPassword']),
+
+    $_loader_publicContext () {
+      // TODO: Can we rely on not being "authenticated" while viewing a public link?
+      // Currently it works. We cannot use publicPage() because that will still return
+      // true when opening the mediaviewer from authenticated routes
+      return !this.isAuthenticated
+    }
+
+  },
+
+  data () {
+    return {
+      $_loader_folderLoading: true
```

Comment:
I don't really like that we have data in a mixin. But since there's a ticket from @LukasHirt now that reminds us to solve this in a better way, I'm ok with merging this anyway.

Reply:
** using the data from the mixin directly.

Reply:
It's especially bad if the consumer needs to access this with `$data`. I assume it'd be a breaking api change to make it a computed property then. Better to make it a computed property from the started. Added an additional `_internal_` infix to the data member to make it obvious it's not meant to be used from outside the mixin itself.

Reply:
Hehehe, I wrote my comment without reloading - so I didn't see your second comment. 
Yep, full ack.

Reply:
Awesome, thanks! 👍